src/extension/ios/iOSDebugModeManager.ts

```diff
@@ -30,37 +30,31 @@ export class IOSDebugModeManager extends ApplePlatformDebugModeManager {
 
         // Find the plistFile with the configuration setting
         // There is a race here between us checking for the plist file, and the application starting up.
-        return this.findPListFileWithRetry(configuration, productName).then((plistFile: string) => {
-            // Set the executorClass to be RCTWebSocketExecutor so on the next startup it will default into debug mode
-            // This is approximately equivalent to clicking the "Debug in Chrome" button
-
-            return (enable
-                ? plistBuddy.setPlistProperty(
-                      plistFile,
-                      IOSDebugModeManager.EXECUTOR_CLASS_SETTING_NAME,
-                      IOSDebugModeManager.WEBSOCKET_EXECUTOR_NAME,
-                  )
-                : plistBuddy.deletePlistProperty(
-                      plistFile,
-                      IOSDebugModeManager.EXECUTOR_CLASS_SETTING_NAME,
-                  )
-            ).then(() =>
-                plistBuddy.setPlistBooleanProperty(
-                    plistFile,
-                    IOSDebugModeManager.REMOTE_DEBUGGING_SETTING_NAME,
-                    enable,
-                ),
-            );
-        });
+        const plistFile = await this.findPListFileWithRetry(configuration, productName);
+        await (enable
```

Comment:
```suggestion
        // Set the executorClass to be RCTWebSocketExecutor so on the next startup it will default into debug mode
        // This is approximately equivalent to clicking the "Debug in Chrome" button
        await (enable
```